pytorch_toolkit/formula_recognition/README.md

```diff
@@ -0,0 +1,158 @@
+    * `disable_layer_3` and `disable_layer_4` - disables layer 3 and 4 in resnet-like backbone
+    * `enable_last_conv` - enables additional convolution layer to adjust number of output channels to the number of input channels in the LSTM
+    * `in_lstm_ch` - number of input LSTM channels, used for `last_conv`
+- `backbone_type`: `resnet` for resnet-like backbone or anything else for original backbone from [im2markup](https://arxiv.org/pdf/1609.04938.pdf) paper
+- `batch_size` - batch size used for training
+- `device` - device for training, used in pytorch .to() method. Possible options: 'cuda', 'cpu', etc. `cpu` is used by default.
+- `head` - configuration of the text recognition head
+    * `beam_width` - witdth used in beam search. 0 - do not use beam search, 1 and more - use beam search with corresponding number of possible tracks.
+    * `dec_rnn_h` - number of channels in decoding
+    * `emb_size` - dimension of the embedding
+    * `enc_rnn_h` - number of channels in encoding
+    * `in_lstm_ch` - number of input in lstm channels, should be equal to `backbone_config.in_lstm_ch`
+    * `max_len` - maximum possible length of the predicted formula
+    * `n_layer` - describe
+- `learning_rate` - learining rate
+- `log_path` - path to store training logs
+- `model_path` - path for model (if one wants to aftertune model)
+- `optimizer` - Adam or SGD
+- `save_dir` - dir to save checkpoints
+- `train_paths` - list of paths from where get training data (if more than one path is specified, datasets are concatenated). If one wants to concatenate more than one instance of the desirable dataset, this dataset should be specified several times.
+- `val_path` - path for validation data
+- `vocab_path` - path where vocab file is stored
+- `train_transforms_list` and
+- `val_transforms_list` - here you can describe set of desirable transformations for train and validation datasets respectively
+- `epochs` - number of epochs to train
+- `clip_grad` - maximum possible value for gradient
+- `old_model` - use this flag if you want to traing model trained in the previous versions of this framework.
+
+
+One can point to pre-trained model [checkpoint](https://download.01.org/opencv/openvino_training_extensions/models/text_spotter/model_step_200000.pth) inside configuration file to start training from pre-trained weights. Change `configs/train_config.yml`:
+```
+...
+model_path: <path_to_weights>
+...
+```
+
+If the model was marked `old_model`, that means the model was trained in older version of this framework (concretely, model checkpoint keys are different from keys used in model now), so if you want to use this model in any context, point out this fact in `config`:
+```
+...
+old_model: true
+...
+```
+
+
+## Evaluation
+
+`tools/test.py` script is designed for quality evaluation of im2latex models.
+
+### PyTorch
+
+Config file of the evaluation is similiar to train config:
+
+```bash
+python tools/test.py --config configs/eval_config.yml
```

Comment:
Where should we write the path to the checkpoint? to the `model_path` field? or in another field? could you write it?

Reply:
The names of the parameters in the evaluation config are the same as in train one, so there is no need to duplicate them. Anyway, all the four configs will be merged into one file.